src/Test/QuickCheck/StateMachineModel.hs

```diff
@@ -36,26 +40,33 @@ import           Text.PrettyPrint.ANSI.Leijen            (Pretty, align, dot,
 
 import           Test.QuickCheck.StateMachineModel.Utils
 
+import           Unsafe.Coerce
+
 ------------------------------------------------------------------------
 
 liftSem
   :: forall cmd resp ix ref m
   .  (Monad m, Enum ix, Typeable ref, RefKit cmd)
   => Typeable resp
+  => Ord (ix, Int)
   => Show resp
   => (cmd resp ref -> m resp)
-  -> cmd resp ix -> StateT [ref] m resp
-liftSem sem cmd = do
+  -> Int
```

Comment:
Could we introduce an
```
type Pid = Int
```
Even if we don't expose it to the user it sure is nice to have some documentation, a `newtype` could also work.